rpc/src/server/mod.rs

```diff
@@ -181,13 +183,26 @@ pub fn spawn_server(
                                 }
                                 _ => {
                                     if allowed_methods.contains(request_method) {
+                                        let uri_string = req.uri().to_string();
+                                        let cache = env.cache.clone();
+                                        if let Some(cached_response) = cache.read().unwrap().get(&uri_string).cloned() {
+                                            return Ok(Response::new(CacheBody::Ready(cached_response)));
+                                        }
                                         let params: Params = params.into_iter().map(|(param, value)| (param.to_string(), value.to_string())).collect();
                                         let query: Query = req.uri().query().map(parse_query_string).unwrap_or_else(HashMap::new);
 
                                         let handler = handler.clone();
                                         let fut = handler(req, params, query, env);
                                         match Pin::from(fut).await {
-                                            Ok(response) => Ok(response),
+                                            Ok(response) => {
+                                                // TODO: more conditions for cache
```

Comment:
Maybe we should not cache certain requests

Reply:
Yes.We will also need time to live for endpoints some endpoints